compiler/runtime.go

```diff
@@ -54,17 +33,8 @@ func (c compiledFn) validate(input values.Object) error {
 		return errors.New("mismatched parameters and properties")
 	}
 	for k, v := range sig.Parameters {
-		if properties[k] != v {
-			// TODO(adam): we should have more thorough and specific type checking to make this work instead of just special casing objects.
-			if v.Nature() == semantic.Object && properties[k].Nature() == semantic.Object {
-				for f := range v.Properties() {
-					if _, ok := properties[k].Properties()[f]; !ok {
-						return fmt.Errorf("parameter %q has the wrong type, expected %v got %v", k, v, properties[k])
-					}
-				}
-			} else {
-				return fmt.Errorf("parameter %q has the wrong type, expected %v got %v", k, v, properties[k])
-			}
+		if !values.AssignableTo(properties[k], v) {
```

Comment:
This is just a precaution right? If this were to return false, it would never have compiled in the first place, right?

Reply:
That's been my impression. This code works fine when you comment out this entire section, but I don't know if that's a good type system or a lack of test coverage.